iceoryx_posh/test/integrationtests/test_service_discovery.cpp

```diff
@@ -596,7 +597,7 @@ TEST_F(ServiceDiscovery_test, FindServiceReturnsMaxServices)
 
     auto serviceContainer = sut.findService(SERVICE, iox::capro::Wildcard, iox::capro::Wildcard);
 
-    EXPECT_THAT(serviceContainer.size(), Eq(iox::MAX_NUMBER_OF_SERVICES));
+    EXPECT_THAT(serviceContainer.size(), Eq(NUM_PUBLISHERS));
```

Comment:
Something weird happens here, we do not find internal publishers or so it seems. Why?

Reply:
@MatthiasKillat I don't understand your comment. The internal `popo::Publishers` are created by the `RouDiEnvironment` have have to be subtracted like you did in line 585.

Reply:
I think they should be found by discovery but apparently they are not.

Reply:
Fixed, we were not searching for the service wildcard. Now it is a proper limit test that finds the maximum services the system can handle, including the internal ones (6 as of now, introspection and service registry).